### PR TITLE
Fix 'releaseGrab' not being called when using grip buttons

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -2340,6 +2338,7 @@ function MyController(hand) {
     };
 
     this.release = function() {
+        this.callEntityMethodOnGrabbed("releaseEquip");
```
Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: is this going to cause releaseEquip on things that were merely grabbed?

Comment (huffman) [inline scripts/system/controllers/handControllerGrab.js]: I'm not sure, I can look into that.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 939d04ef12b320eccb695ed018ca6bfec79ce7c3 available at: https://gist.github.com/39ec93aee0ee764a0060a856c59b3572<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9127-939d04e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9127-939d04e.dmg

File: scripts/system/controllers/handControllerGrab.js [hunk 2]
```diff
@@ -2344,6 +2342,9 @@ function MyController(hand) {
 
         var noVelocity = false;
         if (this.grabbedEntity !== null) {
+            if (this.state === STATE_HOLD) {
+                this.callEntityMethodOnGrabbed("releaseEquip");
+            }
```
Comment (huffman) [inline scripts/system/controllers/handControllerGrab.js]: @seth thoughts?

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa1e813d72000e93930cc0ca99cdbc056ada1728 available at: https://gist.github.com/39ec93aee0ee764a0060a856c59b3572<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9127-aa1e813.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9127-aa1e813.dmg

File: scripts/system/controllers/handControllerGrab.js (hunk 2 above)
Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: looks potentially correct.

Comment (huffman) [inline scripts/system/controllers/handControllerGrab.js]: I was thinking the same thing

**Comment by ZappoMan** [comment]
@huffman - is this ready for testing? can you write a test plan?

**Comment by huffman** [comment]
This change fixes an issue with `releaseEquip` not always being called on the currently equipped entity when unequipping.

To test:

 1. Test that grabbing still works as expected.
 1. Run `https://rawgit.com/huffman/hifi/fix/bow/bow/bow/spawnBow.js` - it should spawn a bow next to you
 1. Confirm that you can:
   1. Equip the bow and shoot it
   1. The arrow shoots as expected
   1. You can unequip the bow using the unequip button on your controllers (on a vive this is the grip button). The bow can still be grabbed, far grabbed, and equipped afterwards.
    1. You can unequip the bow by tilting your hand and pressing the trigger (the "old" way of unequipping). The bow can still be grabbed, far grabbed, and equipped afterwards.

**Comment by Nex-Pro** [comment]
Seems ok but at one point when other users were experiencing frame rate drops  I had some difficulty dropping the bow. I was able to drag it off using other hand.
Mostly ok tho.

Equipped and dropped more than 100 times.

**Pull request merged**